### PR TITLE
Add range to fetch_file function

### PR DESCRIPTION
**Pull request by rositariahi**

### CODE REVIEW

File: xocto/storage/storage.py [hunk 1]
```diff
@@ -568,8 +570,12 @@ def get_key(self, key_path: str, version_id: str | None = None) -> S3Object:
...
         return self._get_boto_object_for_key(key=key_path).content_type
 
-    def fetch_file(self, key_path: str, version_id: str | None = None) -> StreamingBody:
+    def fetch_file(
+        self, key_path: str, version_id: str | None = None, range: str | None = None
+    ) -> StreamingBody:
         boto_object = self._get_boto_object_for_key(key=key_path, version_id=version_id)
+        if range:
+            return boto_object.get(Range=range)["Body"]
         return boto_object.get()["Body"]
```
Comment (IRepeva) [inline xocto/storage/storage.py:579]: 🐼 I'd simplify it to `boto_object.get(Range=range)["Body"]`

Comment (rositariahi) [reply xocto/storage/storage.py:579]: I'd changed it to satisfy mypy - shoud I ignore instead?

Comment (lewis-baker-octopus) [reply xocto/storage/storage.py:579]: I think its fine, better not to execute that code if range is `None`

**Review by lewis-baker-octopus — APPROVED** [review]
Looks good to me 🚀

File: xocto/storage/storage.py (hunk 1 above)
Comment (IRepeva) [reply xocto/storage/storage.py:579]: > I'd changed it to satisfy mypy - shoud I ignore instead?
Better to leave it as it is :)

**Review by IRepeva — APPROVED** [review]
LGTM! 🚀

**Review by hananhafez — APPROVED** [review]
LGTM 🚀

**Pull request merged**